iceoryx_hoofs/test/moduletests/test_unix_domain_sockets.cpp

```diff
@@ -193,99 +196,159 @@ TEST_F(UnixDomainSocket_test, SendOnServerLeadsToError)
+        {""}, [&](auto& msg) { return client.send(msg); }, [&]() { return server.receive(); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfEmptyMessageWithTimedSendAndReceive)
 {
     successfulSendAndReceive(
-        "", [&](auto& msg) { return client.timedSend(msg, 1_ms); }, [&]() { return server.receive(); });
+        {""}, [&](auto& msg) { return client.timedSend(msg, 1_ms); }, [&]() { return server.receive(); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfEmptyMessageWithTimedSendAndTimedReceive)
 {
     successfulSendAndReceive(
-        "", [&](auto& msg) { return client.timedSend(msg, 1_ms); }, [&]() { return server.timedReceive(1_ms); });
+        {""}, [&](auto& msg) { return client.timedSend(msg, 1_ms); }, [&]() { return server.timedReceive(1_ms); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfEmptyMessageWithSendAndTimedReceive)
 {
     successfulSendAndReceive(
-        "", [&](auto& msg) { return client.send(msg); }, [&]() { return server.timedReceive(1_ms); });
+        {""}, [&](auto& msg) { return client.send(msg); }, [&]() { return server.timedReceive(1_ms); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfMaxLengthMessageWithSendAndReceive)
 {
     successfulSendAndReceive(
-        std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x'),
+        {std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x')},
         [&](auto& msg) { return client.send(msg); },
         [&]() { return server.receive(); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfMaxLengthMessageWithTimedSendAndReceive)
 {
     successfulSendAndReceive(
-        std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x'),
+        {std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x')},
         [&](auto& msg) { return client.timedSend(msg, 1_ms); },
         [&]() { return server.receive(); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfMaxLengthMessageWithTimedSendAndTimedReceive)
 {
     successfulSendAndReceive(
-        std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x'),
+        {std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x')},
         [&](auto& msg) { return client.timedSend(msg, 1_ms); },
         [&]() { return server.timedReceive(1_ms); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfMaxLengthMessageWithSendAndTimedReceive)
 {
     successfulSendAndReceive(
-        std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x'),
+        {std::string(UnixDomainSocket::MAX_MESSAGE_SIZE, 'x')},
+        [&](auto& msg) { return client.send(msg); },
+        [&]() { return server.timedReceive(1_ms); });
+}
+
+TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfMultipleMessagesWithSendAndReceive)
+{
+    successfulSendAndReceive(
+        {"Famous hypntoad alike creators from around the world:",
+         "Zoich, proposed mascot for the winter olympics 2014",
+         "Ed Bighead",
+         "Jason Funderberker"},
```

Comment:
I think you mean Jason Funderburker, otherwise it's blasphemy.

Reply:
Oh my god you are right ... I am terribly sorry

Reply:
With ten "Hail Hypnotoad" you will receive forgiveness.

Reply:
Oh no, you misspelled hypnotoad in line 309... 100 Hail Hypnotoad!